winch/codegen/src/regset.rs

```diff
@@ -32,18 +32,55 @@ impl IndexMut<RegClass> for RegSet {
     }
 }
 
+/// Bitset for a particular register class.
+pub struct RegBitSet {
+    /// The register class.
+    class: RegClass,
+    /// The set of allocatable
+    allocatable: u32,
```

Comment:
It's probably a good idea to future-proof this a bit by using `u64`s; all ISAs we support today have 32 GPRs or less but this would be a surprising footgun if we ever exceeded that.

Reply:
Yeah, good call, I've made this a `u64` and introduced a concept of `max` in the `RegBitSet`.